frontend/packages/kubevirt-plugin/integration-tests/tests/vm.snapshots.scenario.ts

```diff
@@ -0,0 +1,53 @@
+import { browser, ExpectedConditions as until } from 'protractor';
+import { testName } from '@console/internal-integration-tests/protractor.conf';
+import { createResource, deleteResource, click } from '@console/shared/src/test-utils/utils';
+import * as editVMSnapshotsView from '../views/vm.snapshots.view';
+import { saveButton, deleteButton } from '../views/kubevirtUIResource.view';
+import { VM_CREATE_AND_EDIT_TIMEOUT_SECS } from './utils/constants/common';
+import { VirtualMachine } from './models/virtualMachine';
+import { getVMManifest } from './mocks/mocks';
+import { getRandStr } from './utils/utils';
+import { READY } from '../../src/utils/strings';
+
+describe('KubeVirt VM Snapshots', () => {
+  const testVM = getVMManifest('Container', testName, `snapshotresourcevm-${getRandStr(5)}`);
+  const vm = new VirtualMachine(testVM.metadata);
+
+  beforeAll(async () => {
+    createResource(testVM);
+  });
+
+  afterAll(() => {
+    deleteResource(testVM);
+  });
+
+  it(
+    'ID() create, restore and delete a snapshot',
+    async () => {
+      await vm.navigateToSnapshots();
+      await click(editVMSnapshotsView.addSnapshotBtn);
+      await editVMSnapshotsView.snapshotNameInput
+        .clear()
+        .then(() => editVMSnapshotsView.snapshotNameInput.sendKeys('test-snapshot'));
+      await click(saveButton);
+
+      // wait for snapshot to be ready
+      await browser.wait(
+        until.textToBePresentInElement(
+          editVMSnapshotsView.getStatusElement('test-snapshot'),
+          READY,
+        ),
+      );
+
+      // TODO: add restore testing (right now not available in the API)
+
+      await click(editVMSnapshotsView.kebab);
```

Comment:
can we use listViewAction instead?

Reply:
`listPage.rows.clickKebabAction(name, deleteKind(kind, humanizeKind));`